<em>Schema metadata produced by ddltrans on Fri May 30 09:35:47 EDT 2003</em>

<html>
<h2>acquisition</h2>
changed some of the pk/fk names to conform to chado conventions (removed _) changed field names (removed _s) changed some tablenames by adding _ to link tables, dropping _ where no link. dropped trailing '-imp' off some tablenames dropped external_database_release_id fields mapped contact links to author table mapped bibliographic references to pub table source_id changed to dbxref_id ok drop table acquisition;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>uri</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>channel_id</td><td>int</td><td>yes</td><td></td><td>channel.channel_id</td></tr>
<tr><td>acquisition_id</td><td>serial</td><td>no</td><td></td><td><li>acquisitionparam.acquisition_id<li>relatedacquisition.associatedacquisition_id<li>relatedacquisition.acquisition_id<li>quantification.acquisition_id</td></tr>
<tr><td>acquisitiondate</td><td>date</td><td>yes</td><td></td><td></td></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
<tr><td>protocol_id</td><td>int</td><td>yes</td><td></td><td>protocol.protocol_id</td></tr>
</table>

<h2>acquisitionparam</h2>
ok drop table if exists acquisitionparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(50)</td><td>no</td><td></td><td></td></tr>
<tr><td>acquisition_id</td><td>int</td><td>no</td><td></td><td>acquisition.acquisition_id</td></tr>
<tr><td>acquisitionparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>analysis</h2>
ok drop table if exists analysis;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>analysis_id</td><td>serial</td><td>no</td><td></td><td><li>analysisfeature.analysis_id<li>analysisprop.analysis_id<li>analysisimplementation.analysis_id</td></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>analysisfeature</h2>
computational analyses generate features (eg genscan generates  transcripts and exons; sim4 alignments generate similarity/match  features)  analysisfeatures are stored using the feature table from  the sequence module. the analysisfeature table is used to  decorate these features, with analysis specific attributes.   a feature is an analysisfeature if and only if there is  a corresponding entry in the analysisfeature table   analysisfeatures will have two or more featureloc entries,  with rank indicating query/subject   analysis_id:     scoredsets are grouped into analyses    rawscore:     this is the native score generated by the program; for example,     the bitscore generated by blast, sim4 or genscan scores.     one should not assume that high is necessarily better than low.    normscore:     this is the rawscore but semi-normalized. complete normalization     to allow comparison of features generated by different programs     would be nice but too difficult. instead the normalization should     strive to enforce the following semantics:      * normscores are floating point numbers >= 0     * high normscores are better than low one.      for most programs, it would be sufficient to make the normscore     the same as this rawscore, providing these semantics are     satisfied.    significance:     this is some kind of expectation or probability metric,     representing the probability that the scoredset would appear     randomly given the model.     as such, any program or person querying this table can assume     the following semantics:      * 0 <= significance <= n, where n is a positive number, theoretically        unbounded but unlikely to be more than 10      * low numbers are better than high numbers.    identity:     percent identity between the locations compared    note that these 4 metrics do not cover the full range of scores   possible; it would be undesirable to list every score possible, as   this should be kept extensible. instead, for non-standard scores, use   the scoredsetprop table.
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>analysis_id</td><td>int</td><td>no</td><td></td><td>analysis.analysis_id</td></tr>
<tr><td>significance</td><td>double</td><td>yes</td><td></td><td></td></tr>
<tr><td>rawscore</td><td>double</td><td>yes</td><td></td><td></td></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>normscore</td><td>double</td><td>yes</td><td></td><td></td></tr>
<tr><td>identity</td><td>double</td><td>yes</td><td></td><td></td></tr>
<tr><td>analysisfeature_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>analysisimplementation</h2>
ok drop table if exists analysisimplementation;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>analysis_id</td><td>int</td><td>no</td><td></td><td>analysis.analysis_id</td></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>analysisimplementation_id</td><td>serial</td><td>no</td><td></td><td><li>analysisimplementationparam.analysisimplementation_id<li>analysisinvocation.analysisimplementation_id</td></tr>
</table>

<h2>analysisimplementationparam</h2>
ok drop table if exists analysisimplementationparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>analysisimplementation_id</td><td>int</td><td>no</td><td></td><td>analysisimplementation.analysisimplementation_id</td></tr>
<tr><td>analysisimplementationparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>analysisinput</h2>
ok drop table if exists analysisinput;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tableinfo_id</td><td>int</td><td>yes</td><td></td><td>tableinfo.tableinfo_id</td></tr>
<tr><td>analysisinput_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>analysisinvocation_id</td><td>int</td><td>no</td><td></td><td>analysisinvocation.analysisinvocation_id</td></tr>
<tr><td>inputrow_id</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>inputvalue</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>analysisinvocation</h2>
ok drop table if exists analysisinvocation;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>analysisinvocation_id</td><td>serial</td><td>no</td><td></td><td><li>analysisinput.analysisinvocation_id<li>analysisinvocationparam.analysisinvocation_id<li>analysisoutput.analysisinvocation_id</td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>analysisimplementation_id</td><td>int</td><td>no</td><td></td><td>analysisimplementation.analysisimplementation_id</td></tr>
</table>

<h2>analysisinvocationparam</h2>
ok drop table if exists analysisinvocationparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>analysisinvocation_id</td><td>int</td><td>no</td><td></td><td>analysisinvocation.analysisinvocation_id</td></tr>
<tr><td>analysisinvocationparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>analysisoutput</h2>
ok drop table if exists analysisoutput;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>analysisoutput_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>type</td><td>varchar(50)</td><td>no</td><td></td><td></td></tr>
<tr><td>analysisinvocation_id</td><td>int</td><td>no</td><td></td><td>analysisinvocation.analysisinvocation_id</td></tr>
<tr><td>value</td><td>int</td><td>no</td><td></td><td></td></tr>
</table>

<h2>analysisprop</h2>
analyses can have various properties attached - eg the parameters  used in running a blast
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>analysis_id</td><td>int</td><td>no</td><td></td><td>analysis.analysis_id</td></tr>
<tr><td>analysisprop_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>pval</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>pkey_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>array</h2>
ok drop table if exists array;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>element_dimensions</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>num_sub_rows</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>version</td><td>varchar(50)  null</td><td>yes</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>substrate_type_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>array_id</td><td>serial</td><td>no</td><td></td><td><li>element.array_id<li>assay.array_id<li>arrayannotation.array_id</td></tr>
<tr><td>num_array_columns</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>platformtype_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>num_grid_columns</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>array_dimensions</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>num_array_rows</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>num_sub_columns</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>manufacturer_id</td><td>int</td><td>no</td><td></td><td>author.author_id</td></tr>
<tr><td>number_of_elements</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>protocol_id</td><td>int</td><td>yes</td><td></td><td>protocol.protocol_id</td></tr>
<tr><td>num_grid_rows</td><td>int</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>arrayannotation</h2>
ok drop table if exists arrayannotation;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(500)</td><td>no</td><td></td><td></td></tr>
<tr><td>array_id</td><td>int</td><td>no</td><td></td><td>array.array_id</td></tr>
<tr><td>value</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>arrayannotation_id</td><td>int</td><td>no</td><td></td><td></td></tr>
</table>

<h2>assay</h2>
ok drop table if exists assay;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>arrayidentifier</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>arraybatchidentifier</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>operator_id</td><td>int</td><td>no</td><td></td><td>author.author_id</td></tr>
<tr><td>array_id</td><td>int</td><td>no</td><td></td><td>array.array_id</td></tr>
<tr><td>assay_id</td><td>serial</td><td>no</td><td></td><td><li>control.assay_id<li>assay_labeledextract.assay_id<li>assay_biomaterial.assay_id<li>studydesign_assay.assay_id<li>studyfactorvalue.assay_id<li>study_assay.assay_id<li>acquisition.assay_id</td></tr>
<tr><td>protocol_id</td><td>int</td><td>yes</td><td></td><td>protocol.protocol_id</td></tr>
<tr><td>assaydate</td><td>date</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>assay_biomaterial</h2>
ok drop table if exists assay_biomaterial;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>biomaterial_id</td><td>int</td><td>no</td><td></td><td>biomaterial.biomaterial_id</td></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
<tr><td>assay_biomaterial_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>assay_labeledextract</h2>
ok drop table if exists assay_labeledextract;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>labeledextract_id</td><td>int</td><td>no</td><td></td><td>biomaterial.biomaterial_id</td></tr>
<tr><td>assay_labeledextract_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>channel_id</td><td>int</td><td>no</td><td></td><td>channel.channel_id</td></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
</table>

<h2>author</h2>
using the FB author table columns
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>surname</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>suffix</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>givennames</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>author_id</td><td>serial</td><td>no</td><td></td><td><li>assay.operator_id<li>wwwuser_author.author_id<li>pub_author.author_id<li>quantification.operator_id<li>biomaterial.biosourceprovider_id<li>study.contact_id<li>array.manufacturer_id</td></tr>
</table>

<h2>biomaterial</h2>
ok renamed from biomaterialimp drop table if exists biomaterial;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>biosourceprovider_id</td><td>int</td><td>yes</td><td></td><td>author.author_id</td></tr>
<tr><td>biomaterial_id</td><td>serial</td><td>no</td><td></td><td><li>assay_labeledextract.labeledextract_id<li>assay_biomaterial.biomaterial_id<li>biomaterialmeasurement.biomaterial_id<li>treatment.biomaterial_id<li>biomaterial_cvterm.biomaterial_id</td></tr>
<tr><td>taxon_id</td><td>int</td><td>yes</td><td></td><td>organism.organism_id</td></tr>
<tr><td>dbxref_id</td><td>varchar(50)</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>string1</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>string2</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>subclass_view</td><td>varchar(27)</td><td>no</td><td></td><td></td></tr>
<tr><td>labelmethod_id</td><td>int</td><td>yes</td><td></td><td>labelmethod.labelmethod_id</td></tr>
</table>

<h2>biomaterial_cvterm</h2>
ok renamed biomaterialcharacteristic to biomaterial_cvterm drop table if exists biomaterial_cvterm;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>value</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>biomaterial_id</td><td>int</td><td>no</td><td></td><td>biomaterial.biomaterial_id</td></tr>
<tr><td>biomaterial_cvterm_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>biomaterialmeasurement</h2>
ok drop table if exists biomaterialmeasurement;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>treatment_id</td><td>int</td><td>no</td><td></td><td>treatment.treatment_id</td></tr>
<tr><td>biomaterialmeasurement_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>biomaterial_id</td><td>int</td><td>no</td><td></td><td>biomaterial.biomaterial_id</td></tr>
<tr><td>unittype_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>channel</h2>
ok drop table if exists channel;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>definition</td><td>varchar(500)</td><td>no</td><td></td><td></td></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>channel_id</td><td>serial</td><td>no</td><td></td><td><li>labelmethod.channel_id<li>assay_labeledextract.channel_id<li>acquisition.channel_id</td></tr>
</table>

<h2>compositeelementresult</h2>
dropped compositeelementannotation.  use featureprop instead. dropped compositeelementgus.         use feature instead. dropped compositeelement.            use feature instead. ok drop table if exists compositeelementresult;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tinyint1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinyint2</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinyint3</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>float1</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>string1</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>compositeelementresult_id</td><td>serial</td><td>no</td><td></td><td><li>elementresult.compositeelementresult_id</td></tr>
<tr><td>float2</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>string2</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>compositeelement_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>float3</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float4</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>quantification_id</td><td>int</td><td>no</td><td></td><td>quantification.quantification_id</td></tr>
<tr><td>char1</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char2</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char3</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint2</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint3</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>subclass_view</td><td>varchar(27)</td><td>no</td><td></td><td></td></tr>
<tr><td>int1</td><td>int</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>control</h2>
ok drop table if exists control;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tableinfo_id</td><td>int</td><td>no</td><td></td><td>tableinfo.tableinfo_id</td></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>row_id</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>control_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
<tr><td>controltype_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>cv</h2>
- term - term_definition - term2term - graph_path - term_synonym - term_dbxref -- dbxref - tricky, namespace clash...  The cvterm module design is based on the ontology
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>cvdefinition</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>cvname</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>cv_id</td><td>serial</td><td>no</td><td></td><td><li>cvpath.cv_id<li>cvterm.cv_id</td></tr>
</table>

<h2>cvpath</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pathdistance</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>subjterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>reltype_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>cvpath_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>cv_id</td><td>int</td><td>no</td><td></td><td>cv.cv_id</td></tr>
<tr><td>objterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>cvrelationship</h2>
the primary dbxref for this term.  Other dbxrefs may be cvterm_dbxref  The unique key on termname, cv_id ensures that all terms are  unique within a given cv
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>subjterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>reltype_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>objterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>cvrelationship_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>cvterm</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>termdefinition</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>name</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>cv_id</td><td>int</td><td>no</td><td></td><td>cv.cv_id</td></tr>
<tr><td>cvterm_id</td><td>serial</td><td>no</td><td></td><td><li>control.controltype_id<li>pubprop.pkey_id<li>element.element_type_id<li>cvterm_dbxref.cvterm_id<li>feature.type_id<li>cvtermsynonym.cvterm_id<li>protocol.protocol_type_id<li>wwwuser_cvterm.cvterm_id<li>pub_relationship.type_id<li>feature_cvterm.cvterm_id<li>phenotype.statement_type<li>cvpath.subjterm_id<li>cvpath.reltype_id<li>cvpath.objterm_id<li>cvrelationship.subjterm_id<li>cvrelationship.reltype_id<li>cvrelationship.objterm_id<li>phenotype_cvterm.cvterm_id<li>featureprop.pkey_id<li>studydesigndescription.descriptiontype_id<li>dbxrefprop.pkey_id<li>synonym.type_id<li>analysisprop.pkey_id<li>biomaterialmeasurement.unittype_id<li>treatment.treatmenttype_id<li>biomaterial_cvterm.cvterm_id<li>feature_relationship.type_id<li>expression_cvterm.cvterm_id<li>protocolparam.datatype_id<li>protocolparam.unittype_id<li>processimplementation.processtype_id<li>pub.type_id<li>studydesign.studydesigntype_id<li>studyfactor.studyfactortype_id<li>processresult.unittype_id<li>array.substrate_type_id<li>array.platformtype_id</td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
</table>

<h2>cvterm_dbxref</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>dbxref_id</td><td>int</td><td>no</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>cvterm_dbxref_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>cvtermsynonym</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>cvtermsynonym_id</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>termsynonym</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>db</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>db_id</td><td>varchar(255)</td><td>no</td><td></td><td><li>dbxref.dbname</td></tr>
<tr><td>url</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>dbxref</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>accession</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>version</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>dbxrefdescription</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>dbname</td><td>varchar(255)</td><td>no</td><td></td><td>db.db_id</td></tr>
<tr><td>dbxref_id</td><td>serial</td><td>no</td><td></td><td><li>element.dbxref_id<li>cvterm_dbxref.dbxref_id<li>feature.dbxref_id<li>protocol.dbxref_id<li>assay.dbxref_id<li>dbxrefprop.dbxref_id<li>pub_dbxref.dbxref_id<li>feature_dbxref.dbxref_id<li>biomaterial.dbxref_id<li>cvterm.dbxref_id<li>study.dbxref_id<li>organism_dbxref.dbxref_id<li>array.dbxref_id</td></tr>
</table>

<h2>dbxrefprop</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>dbxrefprop_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>pval</td><td>text</td><td>no</td><td></td><td></td></tr>
<tr><td>prank</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>pkey_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>dbxref_id</td><td>int</td><td>no</td><td></td><td>dbxref.dbxref_id</td></tr>
</table>

<h2>eimage</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>eimage_type</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>eimage_data</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>image_uri</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>eimage_id</td><td>serial</td><td>no</td><td></td><td><li>expression_image.eimage_id</td></tr>
</table>

<h2>element</h2>
dropped elementannotation.  use featureprop instead. ok drop table if exists element;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tinyint1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallstring1</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallstring2</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>string1</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>string2</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>array_id</td><td>int</td><td>no</td><td></td><td>array.array_id</td></tr>
<tr><td>tinystring1</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinystring2</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char1</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char2</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char3</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char4</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char5</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char6</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>char7</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>element_id</td><td>serial</td><td>no</td><td></td><td><li>elementresult.element_id</td></tr>
<tr><td>element_type_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>feature_id</td><td>int</td><td>yes</td><td></td><td>feature.feature_id</td></tr>
<tr><td>subclass_view</td><td>varchar(27)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>elementresult</h2>
ok drop table if exists elementresult;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>background</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>string1</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>compositeelementresult_id</td><td>int</td><td>yes</td><td></td><td>compositeelementresult.compositeelementresult_id</td></tr>
<tr><td>string2</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>quantification_id</td><td>int</td><td>no</td><td></td><td>quantification.quantification_id</td></tr>
<tr><td>tinystring1</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>int10</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinystring2</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>int11</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinystring3</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>int12</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>char1</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>int13</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>char2</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>int14</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>char3</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>int15</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>char4</td><td>varchar(5)</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint2</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>element_id</td><td>int</td><td>no</td><td></td><td>element.element_id</td></tr>
<tr><td>background_sd</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallint3</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>subclass_view</td><td>varchar(27)</td><td>no</td><td></td><td></td></tr>
<tr><td>int1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int2</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int3</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int4</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinyint1</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int5</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallstring1</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinyint2</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int6</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>smallstring2</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>tinyint3</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int7</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int8</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>int9</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>float1</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float2</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float3</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float4</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float5</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float6</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float7</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float8</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float9</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>foreground_sd</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>elementresult_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>float10</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float11</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float12</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float13</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>foreground</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
<tr><td>float14</td><td>float(15)</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>expression</h2>
This module is totally dependant on the sequence module.  Objects in the  genetic module cannot connect to expression data except by going via the  sequence module  We assume that we'll *always* have a controlled vocabulary for expression  data.   If an experiment used a set of cv terms different from the ones  FlyBase uses (bodypart cv, bodypart qualifier cv, and the temporal cv  (which is stored in the curaton.doc under GAT6 btw)), they'd have to give  us the cv terms, which we could load into the cv module
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>description</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>expression_id</td><td>serial</td><td>no</td><td></td><td><li>expression_image.expression_id<li>expression_pub.expression_id<li>expression_cvterm.expression_id<li>wwwuser_expression.expression_id<li>feature_expression.expression_id</td></tr>
</table>

<h2>expression_cvterm</h2>
What are the possibities of combination when more than one cvterm is used  in a field?   For eg (in <p> here):   <t> E | early <a> <p> anterior & dorsal  If the two terms used in a particular field are co-equal (both from the  same CV, is the relation always "&"?   May we find "or"?   Obviously another case is when a bodypart term and a bodypart qualifier  term are used in a specific field, eg:    <t> L | third instar <a> larval antennal segment sensilla | subset <p   WRT the three-part <t><a><p> statements, are the values in the different  parts *always* from different vocabularies in proforma.CV?   If not,  we'll need to have some kind of type qualifier telling us whether the  cvterm used is <t>, <a>, or <p>
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>rank</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>expression_id</td><td>int</td><td>no</td><td></td><td>expression.expression_id</td></tr>
<tr><td>expression_cvterm_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>expression_image</h2>
we expect images in eimage_data (eg jpegs) to be uuencoded  describes the type of data in eimage_data
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>eimage_id</td><td>int</td><td>no</td><td></td><td>eimage.eimage_id</td></tr>
<tr><td>expression_id</td><td>int</td><td>no</td><td></td><td>expression.expression_id</td></tr>
<tr><td>expression_image_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>expression_pub</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>expression_pub_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>expression_id</td><td>int</td><td>no</td><td></td><td>expression.expression_id</td></tr>
</table>

<h2>feature</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>timeaccessioned</td><td>timestamp</td><td>no</td><td>timeaccessioned and timelastmodified are for handling object accession/  modification timestamps (as opposed to db auditing info, handled elsewhere).  The expectation is that these fields would be available to software  interacting with chado.</td><td></td></tr>
<tr><td>name</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>timelastmodified</td><td>timestamp</td><td>no</td><td></td><td></td></tr>
<tr><td>residues</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>feature_id</td><td>serial</td><td>no</td><td></td><td><li>element.feature_id<li>wwwuser_feature.feature_id<li>feature_cvterm.feature_id<li>featureloc.srcfeature_id<li>featureloc.feature_id<li>featureprop.feature_id<li>analysisfeature.feature_id<li>feature_genotype.feature_id<li>compositeelementresult.compositeelement_id<li>featurepos.feature_id<li>featurepos.map_feature_id<li>feature_dbxref.feature_id<li>feature_synonym.feature_id<li>featurerange.leftendf_id<li>featurerange.rightendf_id<li>featurerange.feature_id<li>featurerange.leftstartf_id<li>featurerange.rightstartf_id<li>feature_relationship.objfeature_id<li>feature_relationship.subjfeature_id<li>feature_phenotype.feature_id<li>feature_pub.feature_id<li>interaction_subj.feature_id<li>interaction_obj.feature_id<li>feature_expression.feature_id</td></tr>
<tr><td>uniquename</td><td>text</td><td>no</td><td></td><td></td></tr>
<tr><td>seqlen</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>md5checksum</td><td>char(32)</td><td>yes</td><td></td><td></td></tr>
<tr><td>organism_id</td><td>int</td><td>no</td><td></td><td>organism.organism_id</td></tr>
<tr><td>type_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>is_analysis</td><td>boolean</td><td>no</td><td></td><td></td></tr>
</table>

<h2>feature_cvterm</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>feature_cvterm_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>feature_dbxref</h2>
links a feature to dbxrefs.  Note that there is also feature.dbxref_id  link for the primary dbxref link.
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>feature_dbxref_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>is_current</td><td>boolean</td><td>no</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>no</td><td></td><td>dbxref.dbxref_id</td></tr>
</table>

<h2>feature_expression</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>expression_id</td><td>int</td><td>no</td><td></td><td>expression.expression_id</td></tr>
<tr><td>feature_expression_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>feature_genotype</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>feature_genotype_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>genotype_id</td><td>int</td><td>no</td><td></td><td>genotype.genotype_id</td></tr>
</table>

<h2>feature_phenotype</h2>
type of phenotypic statement  [Chris, we need this or something like it  for FB where we have three types of statement in *k: "Phenotypic class:",  "Phenotype manifest in:", and free-text]  Do we want to call this simply genotype_id to allow natural joins?
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>phenotype_id</td><td>int</td><td>no</td><td></td><td>phenotype.phenotype_id</td></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>feature_phenotype_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>feature_pub</h2>
phase: phase of translation wrt srcfeature_id.  Values are 0,1,2
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>feature_pub_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>feature_relationship</h2>
features can be arranged in graphs, eg exon partof transcript  partof gene; translation madeby transcript  if type is thought of as a verb, each arc makes a statement  [SUBJECT VERB OBJECT]  object can also be thought of as parent, and subject as child   we include the relationship rank/order, because even though  most of the time we can order things implicitly by sequence  coordinates, we can't always do this - eg transpliced genes.  it's also useful for quickly getting implicit introns
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_relationship_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>objfeature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>subjfeature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>type_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>relrank</td><td>int</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>feature_synonym</h2>
type_id: types would be symbol and fullname for now  synonym_sgml: sgml-ized version of symbols
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>is_internal</td><td>boolean</td><td>no</td><td></td><td></td></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>synonym_id</td><td>int</td><td>no</td><td></td><td>synonym.synonym_id</td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>feature_synonym_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>is_current</td><td>boolean</td><td>no</td><td></td><td></td></tr>
</table>

<h2>featureloc</h2>
dbxref_id here is intended for the primary dbxref for this feature.  Additional dbxref links are made via feature_dbxref  name: the human-readable common name for a feature, for display  uniquename: the unique name for a feature; may not be particularly human-readable  timeaccessioned and timelastmodified are for handling object accession/  modification timestamps (as opposed to db auditing info, handled elsewhere).  The expectation is that these fields would be available to software  interacting with chado.  each feature can have 0 or more locations.  multiple locations do NOT indicate non-contiguous locations.  instead they designate alternate locations or grouped locations;  for instance, a feature designating a blast hit or hsp will have two  locations, one on the query feature, one on the subject feature.  features representing sequence variation could have alternate locations  instantiated on a feature on the mutant strain.  the field "rank" is used to differentiate these different locations.  the default rank '0' is used for the main/primary location (eg in  similarity features, 0 is query, 1 is subject), although sometimes  this will be symmeytical and there is no primary location.   redundant locations can also be stored - for instance, the position  of an exon on a BAC and in global coordinates. the field "locgroup"  is used to differentiate these groupings of locations. the default  locgroup '0' is used for the main/primary location, from which the  others can be derived via coordinate transformations. another  example of redundant locations is storing ORF coordinates relative  to both transcript and genome. redundant locations open the possibility  of the database getting into inconsistent states; this schema gives  us the flexibility of both 'warehouse' instantiations with redundant  locations (easier for querying) and 'management' instantiations with  no redundant locations.  most features (exons, transcripts, etc) will have 1 location, with  locgroup and rank equal to 0   an example of using both locgroup and rank:  imagine a feature indicating a conserved region between the chromosomes  of two different species. we may want to keep redundant locations on  both contigs and chromosomes. we would thus have 4 locations for the  single conserved region feature - two distinct locgroups (contig level  and chromosome level) and two distinct ranks (for the two species).  altresidues is used to store alternate residues of a feature, when these  differ from feature.residues. for instance, a SNP feature located on  a wild and mutant protein would have different alresidues.  for alignment/similarity features, the altresidues is used to represent  the alignment string.  note on variation features; even if we don't want to instantiate a mutant  chromosome/contig feature, we can still represent a SNP etc with 2 locations,  one (rank 0) on the genome, the other (rank 1) would have most fields null,  except for altresidues  IMPORTANT: fnbeg and fnend are space-based (INTERBASE) coordinates  this is vital as it allows us to represent zero-length  features eg splice sites, insertion points without  an awkward fuzzy system  nbeg, nend are for feature natural begin/end  by natural begin, end we mean these are the actual  beginning (5' position) and actual end (3' position)  rather than the low position and high position, as  these terms are sometimes erroneously used
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>srcfeature_id</td><td>int</td><td>yes</td><td></td><td>feature.feature_id</td></tr>
<tr><td>phase</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>is_nbeg_partial</td><td>boolean</td><td>no</td><td></td><td></td></tr>
<tr><td>nend</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>featureloc_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>locgroup</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>rank</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>residue_info</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>strand</td><td>smallint</td><td>yes</td><td></td><td></td></tr>
<tr><td>is_nend_partial</td><td>boolean</td><td>no</td><td></td><td></td></tr>
<tr><td>nbeg</td><td>int</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>featuremap</h2>
NOTE: this module is all due for revision...  A possibly problematic case is where we want to localize an object  to the left or right of a feature (but not within it):                       |---------|  feature-to-map         ------------------------------------------------- map                 |------|         |----------|   features to map wrt   How do we map the 3' end of the feature-to-map?  TODO:  Get a comprehensive set of mapping use-cases  one set of use-cases is aberrations (which will all be involved with this  module).   Simple aberrations should be do-able, but what about cases where  a breakpoint interrupts a gene?  This would be an example of the problematic  case above...  (or?)
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>mapdesc</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>mapunit</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>featuremap_id</td><td>serial</td><td>no</td><td></td><td><li>featuremap_pub.featuremap_id<li>featurepos.featuremap_id<li>featurerange.featuremap_id</td></tr>
<tr><td>mapname</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>featuremap_pub</h2>
map_feature_id links to the feature (map) upon which the feature is  being localized
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>featuremap_pub_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>featuremap_id</td><td>int</td><td>no</td><td></td><td>featuremap.featuremap_id</td></tr>
</table>

<h2>featurepos</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>mappos</td><td>float</td><td>no</td><td></td><td></td></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>featurepos_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>map_feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>featuremap_id</td><td>serial</td><td>no</td><td></td><td>featuremap.featuremap_id</td></tr>
</table>

<h2>featureprop</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>featureprop_id</td><td>serial</td><td>no</td><td></td><td><li>featureprop_pub.featureprop_id</td></tr>
<tr><td>pval</td><td>text</td><td>no</td><td></td><td></td></tr>
<tr><td>prank</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>pkey_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>featureprop_pub</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>featureprop_id</td><td>int</td><td>no</td><td></td><td>featureprop.featureprop_id</td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>featureprop_pub_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>featurerange</h2>
In cases where the start and end of a mapped feature is a range, leftendf  and rightstartf are populated.  featuremap_id is the id of the feature being mapped  leftstartf_id, leftendf_id, rightstartf_id, rightendf_id are the ids of  features with respect to with the feature is being mapped.  These may  be cytological bands.
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>leftendf_id</td><td>int</td><td>yes</td><td></td><td>feature.feature_id</td></tr>
<tr><td>featurerange_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>rangestr</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>featuremap_id</td><td>int</td><td>no</td><td></td><td>featuremap.featuremap_id</td></tr>
<tr><td>rightendf_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>leftstartf_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>rightstartf_id</td><td>int</td><td>yes</td><td></td><td>feature.feature_id</td></tr>
</table>

<h2>genotype</h2>
This module depends on the sequence, pub, and cv modules  18-JAN-03 (DE): This module is unfinished and due for schema review (Bill  Gelbart will be leading the charge)
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>description</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>genotype_id</td><td>serial</td><td>no</td><td></td><td><li>phenotype.background_genotype_id<li>wwwuser_genotype.genotype_id<li>feature_genotype.genotype_id<li>interaction.background_genotype_id</td></tr>
</table>

<h2>interaction</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>phenotype_id</td><td>int</td><td>yes</td><td></td><td>phenotype.phenotype_id</td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>description</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>interaction_id</td><td>serial</td><td>no</td><td></td><td><li>wwwuser_interaction.interaction_id<li>interaction_subj.interaction_id<li>interaction_obj.interaction_id</td></tr>
<tr><td>background_genotype_id</td><td>int</td><td>yes</td><td>Do we want to call this simply genotype_id to allow natural joins?</td><td>genotype.genotype_id</td></tr>
</table>

<h2>interaction_obj</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>interaction_id</td><td>int</td><td>no</td><td></td><td>interaction.interaction_id</td></tr>
<tr><td>interaction_obj_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>interaction_subj</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>interaction_id</td><td>int</td><td>no</td><td></td><td>interaction.interaction_id</td></tr>
<tr><td>interaction_subj_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>labelmethod</h2>
ok drop table if exists labelmethod;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>labelmethod</td><td>varchar(1000)</td><td>yes</td><td></td><td></td></tr>
<tr><td>labelused</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>channel_id</td><td>int</td><td>no</td><td></td><td>channel.channel_id</td></tr>
<tr><td>labelmethod_id</td><td>serial</td><td>no</td><td></td><td><li>biomaterial.labelmethod_id</td></tr>
<tr><td>protocol_id</td><td>int</td><td>no</td><td></td><td>protocol.protocol_id</td></tr>
</table>

<h2>magedocumentation</h2>
ok drop table if exists magedocumentation;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tableinfo_id</td><td>int</td><td>no</td><td></td><td>tableinfo.tableinfo_id</td></tr>
<tr><td>magedocumentation_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>row_id</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>mageidentifier</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>mageml_id</td><td>int</td><td>no</td><td></td><td>mageml.mageml_id</td></tr>
</table>

<h2>mageml</h2>
ok  warning - mage_ml does not appear in core.tableinfo drop table if exists mageml;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>mage_ml</td><td>varchar</td><td>no</td><td></td><td></td></tr>
<tr><td>mage_package</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>mageml_id</td><td>serial</td><td>no</td><td></td><td><li>magedocumentation.mageml_id</td></tr>
</table>

<h2>organism</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>genus</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>abbrev</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>comment</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>common_name</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>taxgroup</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>organism_id</td><td>serial</td><td>no</td><td></td><td><li>feature.organism_id<li>wwwuser_organism.organism_id<li>biomaterial.taxon_id<li>organism_dbxref.organism_id</td></tr>
<tr><td>species</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>organism_dbxref</h2>
Compared to mol5..Species, organism table lacks "approved char(1) null".  We need to work w/ Aubrey & Michael to ensure that we don't need this in  future [dave]   in response: this is very specific to a limited use case I think;  if it's really necessary we can have an organismprop table  for adding internal project specific data  [cjm]
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>organism_id</td><td>int</td><td>no</td><td></td><td>organism.organism_id</td></tr>
<tr><td>organism_dbxref_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>no</td><td></td><td>dbxref.dbxref_id</td></tr>
</table>

<h2>phenotype</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>phenotype_id</td><td>serial</td><td>no</td><td></td><td><li>phenotype_cvterm.phenotype_id<li>interaction.phenotype_id<li>wwwuser_phenotype.phenotype_id<li>feature_phenotype.phenotype_id</td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>description</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>background_genotype_id</td><td>int</td><td>yes</td><td></td><td>genotype.genotype_id</td></tr>
<tr><td>statement_type</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>phenotype_cvterm</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>phenotype_id</td><td>int</td><td>no</td><td></td><td>phenotype.phenotype_id</td></tr>
<tr><td>phenotype_cvterm_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>prank</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>processimplementation</h2>
ok drop table if exists processimplementation;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>processimplementation_id</td><td>serial</td><td>no</td><td></td><td><li>processimplementationparam.processimplementation_id<li>processinvocation.processimplementation_id</td></tr>
<tr><td>processtype_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>processimplementationparam</h2>
ok drop table if exists processimplementationparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>processimplementation_id</td><td>int</td><td>no</td><td></td><td>processimplementation.processimplementation_id</td></tr>
<tr><td>processimplementationparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>processinvocation</h2>
ok drop table if exists processinvocation;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>processimplementation_id</td><td>int</td><td>no</td><td></td><td>processimplementation.processimplementation_id</td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>processinvocation_id</td><td>serial</td><td>no</td><td></td><td><li>processio.processinvocation_id<li>processinvocationparam.processinvocation_id<li>processinvocation_quantification.processinvocation_id</td></tr>
<tr><td>processinvocationdate</td><td>date</td><td>no</td><td></td><td></td></tr>
</table>

<h2>processinvocation_quantification</h2>
ok renamed from processinv_quantification to processinvocation_quantification drop table if exists processinvocation_quantification;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>quantification_id</td><td>int</td><td>no</td><td></td><td>quantification.quantification_id</td></tr>
<tr><td>processinvocation_quantification_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>processinvocation_id</td><td>int</td><td>no</td><td></td><td>processinvocation.processinvocation_id</td></tr>
</table>

<h2>processinvocationparam</h2>
ok drop table if exists processinvocationparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>processinvocation_id</td><td>int</td><td>no</td><td></td><td>processinvocation.processinvocation_id</td></tr>
<tr><td>processinvocationparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>processio</h2>
ok drop table if exists processio;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>input_role</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>processio_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>inputrow_id</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>inputtable_id</td><td>int</td><td>no</td><td></td><td>tableinfo.tableinfo_id</td></tr>
<tr><td>outputrow_id</td><td>int</td><td>no</td><td></td><td>processresult.processresult_id</td></tr>
<tr><td>processinvocation_id</td><td>int</td><td>no</td><td></td><td>processinvocation.processinvocation_id</td></tr>
</table>

<h2>processresult</h2>
ok drop table if exists processresult;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>processresult_id</td><td>serial</td><td>no</td><td></td><td><li>processio.outputrow_id</td></tr>
<tr><td>value</td><td>float(15)</td><td>no</td><td></td><td></td></tr>
<tr><td>unittype_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>project</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>project_id</td><td>serial</td><td>no</td><td></td><td><li>wwwuser_project.project_id<li>projectlink.project_id</td></tr>
<tr><td>description</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>projectlink</h2>
ok drop table if exists projectlink;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tableinfo_id</td><td>int</td><td>no</td><td></td><td>tableinfo.tableinfo_id</td></tr>
<tr><td>currentversion</td><td>varchar(4)</td><td>yes</td><td></td><td></td></tr>
<tr><td>projectlink_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>project_id</td><td>int</td><td>no</td><td></td><td>project.project_id</td></tr>
<tr><td>id</td><td>int</td><td>no</td><td></td><td></td></tr>
</table>

<h2>protocol</h2>
ok drop table if exists protocol;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>uri</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>hardwaredescription</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>int</td><td>yes</td><td></td><td>pub.pub_id</td></tr>
<tr><td>protocoldescription</td><td>varchar(4000)</td><td>yes</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
<tr><td>protocol_id</td><td>serial</td><td>no</td><td></td><td><li>labelmethod.protocol_id<li>assay.protocol_id<li>treatment.protocol_id<li>quantification.protocol_id<li>protocolparam.protocol_id<li>acquisition.protocol_id<li>array.protocol_id</td></tr>
<tr><td>softwaredescription</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>protocol_type_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>protocolparam</h2>
ok drop table if exists protocolparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>datatype_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>value</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>protocol_id</td><td>int</td><td>no</td><td></td><td>protocol.protocol_id</td></tr>
<tr><td>unittype_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>protocolparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>pub</h2>
We should take a look in OMG for a standard representation we might use  instead of this.
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pages</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>pyear</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>title</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>serial</td><td>no</td><td></td><td><li>pubprop.pub_id<li>featuremap_pub.pub_id<li>protocol.pub_id<li>featureprop_pub.pub_id<li>pub_relationship.obj_pub_id<li>pub_relationship.subj_pub_id<li>feature_cvterm.pub_id<li>phenotype.pub_id<li>synonym_pub.pub_id<li>expression_pub.pub_id<li>pub_author.pub_id<li>interaction.pub_id<li>pub_dbxref.pub_id<li>feature_synonym.pub_id<li>wwwuser_pub.pub_id<li>feature_pub.pub_id<li>study.pub_id</td></tr>
<tr><td>is_obsolete</td><td>boolean</td><td>yes</td><td></td><td></td></tr>
<tr><td>volume</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>issue</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>miniref</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>volumetitle</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>series_name</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>pubplace</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>publisher</td><td>varchar(255)</td><td>yes</td><td></td><td></td></tr>
<tr><td>type_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>pub_author</h2>
givennames: first name, initials  suffix: Jr., Sr., etc
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>arank</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>pub_author_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>editor</td><td>boolean</td><td>yes</td><td></td><td></td></tr>
<tr><td>author_id</td><td>int</td><td>no</td><td></td><td>author.author_id</td></tr>
</table>

<h2>pub_dbxref</h2>
Handle links to eg, pubmed, biosis, zoorec, OCLC, mdeline, ISSN, coden...
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>pub_dbxref_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>dbxref_id</td><td>int</td><td>no</td><td></td><td>dbxref.dbxref_id</td></tr>
</table>

<h2>pub_relationship</h2>
title: title of paper, chapter of book, journal, etc  volumetitle: title of part if one of a series  series_name: full name of (journal) series  pages: page number range[s], eg, 457--459, viii + 664pp, lv--lvii  type_id: the type of the publication (book, journal, poem, graffiti, etc)  is_obsolete: do we want this even though we have the relationship in pub_relationship?  Handle relationships between publications, eg, when one publication  makes others obsolete, when one publication contains errata with  respect to other publication(s), or when one publication also  appears in another pub (I think these three are it - at least for fb)
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pub_relationship_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>obj_pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>type_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>subj_pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
</table>

<h2>pubprop</h2>
arank: order of author in author list for this pub  editor: indicates whether the author is an editor for linked publication
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>pval</td><td>text</td><td>no</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>prank</td><td>integer</td><td>yes</td><td></td><td></td></tr>
<tr><td>pubprop_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>pkey_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>quantification</h2>
ok drop table if exists quantification;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>uri</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>resulttable_id</td><td>int</td><td>yes</td><td></td><td>tableinfo.tableinfo_id</td></tr>
<tr><td>acquisition_id</td><td>int</td><td>no</td><td></td><td>acquisition.acquisition_id</td></tr>
<tr><td>operator_id</td><td>int</td><td>yes</td><td></td><td>author.author_id</td></tr>
<tr><td>quantification_id</td><td>serial</td><td>no</td><td></td><td><li>relatedquantification.associatedquantification_id<li>relatedquantification.quantification_id<li>compositeelementresult.quantification_id<li>elementresult.quantification_id<li>quantificationparam.quantification_id<li>processinvocation_quantification.quantification_id</td></tr>
<tr><td>quantificationdate</td><td>date</td><td>yes</td><td></td><td></td></tr>
<tr><td>protocol_id</td><td>int</td><td>yes</td><td></td><td>protocol.protocol_id</td></tr>
</table>

<h2>quantificationparam</h2>
ok drop table if exists quantificationparam;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>quantification_id</td><td>int</td><td>no</td><td></td><td>quantification.quantification_id</td></tr>
<tr><td>quantificationparam_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>value</td><td>varchar(50)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>relatedacquisition</h2>
ok drop table if exists relatedacquisition;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>associateddesignation</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>designation</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>associatedacquisition_id</td><td>int</td><td>no</td><td></td><td>acquisition.acquisition_id</td></tr>
<tr><td>acquisition_id</td><td>int</td><td>no</td><td></td><td>acquisition.acquisition_id</td></tr>
<tr><td>relatedacquisition_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>relatedquantification</h2>
ok drop table if exists relatedquantification;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>associatedquantification_id</td><td>int</td><td>no</td><td></td><td>quantification.quantification_id</td></tr>
<tr><td>quantification_id</td><td>int</td><td>no</td><td></td><td>quantification.quantification_id</td></tr>
<tr><td>relatedquantification_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>associateddesignation</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
<tr><td>designation</td><td>varchar(50)</td><td>yes</td><td></td><td></td></tr>
</table>

<h2>study</h2>
ok drop table if exists study;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>contact_id</td><td>int</td><td>no</td><td></td><td>author.author_id</td></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>int</td><td>yes</td><td></td><td>pub.pub_id</td></tr>
<tr><td>description</td><td>varchar(4000)</td><td>yes</td><td></td><td></td></tr>
<tr><td>study_id</td><td>serial</td><td>no</td><td></td><td><li>study_assay.study_id<li>studydesign.study_id</td></tr>
<tr><td>dbxref_id</td><td>int</td><td>yes</td><td></td><td>dbxref.dbxref_id</td></tr>
</table>

<h2>study_assay</h2>
ok renamed from studyassay to study_assay drop table if exists study_assay;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
<tr><td>study_id</td><td>int</td><td>no</td><td></td><td>study.study_id</td></tr>
<tr><td>study_assay_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>studydesign</h2>
ok drop table if exists studydesign;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>studydesigntype_id</td><td>int  null</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>description</td><td>varchar(4000)</td><td>yes</td><td></td><td></td></tr>
<tr><td>study_id</td><td>int</td><td>no</td><td></td><td>study.study_id</td></tr>
<tr><td>studydesign_id</td><td>serial</td><td>no</td><td></td><td><li>studydesign_assay.studydesign_id<li>studydesigndescription.studydesign_id<li>studyfactor.studydesign_id</td></tr>
</table>

<h2>studydesign_assay</h2>
ok renamed from studydesignassay to studydesign_assay drop table if exists studydesign_assay;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
<tr><td>studydesign_id</td><td>int</td><td>no</td><td></td><td>studydesign.studydesign_id</td></tr>
<tr><td>studydesign_assay_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>studydesigndescription</h2>
ok drop table if exists studydesigndescription;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>descriptiontype_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>studydesigndescription_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(4000)</td><td>no</td><td></td><td></td></tr>
<tr><td>studydesign_id</td><td>int</td><td>no</td><td></td><td>studydesign.studydesign_id</td></tr>
</table>

<h2>studyfactor</h2>
ok drop table if exists studyfactor;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>description</td><td>varchar(500)</td><td>yes</td><td></td><td></td></tr>
<tr><td>studyfactor_id</td><td>serial</td><td>no</td><td></td><td><li>studyfactorvalue.studyfactor_id</td></tr>
<tr><td>studyfactortype_id</td><td>int</td><td>yes</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>studydesign_id</td><td>int</td><td>no</td><td></td><td>studydesign.studydesign_id</td></tr>
</table>

<h2>studyfactorvalue</h2>
ok drop table if exists studyfactorvalue;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>studyfactorvalue_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>factorvalue</td><td>varchar(100)</td><td>no</td><td></td><td></td></tr>
<tr><td>studyfactor_id</td><td>int</td><td>no</td><td></td><td>studyfactor.studyfactor_id</td></tr>
<tr><td>assay_id</td><td>int</td><td>no</td><td></td><td>assay.assay_id</td></tr>
</table>

<h2>synonym</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>name</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
<tr><td>synonym_id</td><td>serial</td><td>no</td><td></td><td><li>synonym_pub.synonym_id<li>feature_synonym.synonym_id</td></tr>
<tr><td>type_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>synonym_sgml</td><td>varchar(255)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>synonym_pub</h2>
pub_id: the pub_id link is for relating the usage of a given synonym to the  publication in which it was used  is_current: the is_current bit indicates whether the linked synonym is the  current -official- symbol for the linked feature  is_internal: typically a synonym exists so that somebody querying the db with an  obsolete name can find the object they're looking for (under its current  name.  If the synonym has been used publicly & deliberately (eg in a  paper), it my also be listed in reports as a synonym.   If the synonym  was not used deliberately (eg, there was a typo which went public), then  the is_internal bit may be set to 'true' so that it is known that the  synonym is "internal" and should be queryable but should not be listed  in reports as a valid synonym.
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>synonym_id</td><td>int</td><td>no</td><td></td><td>synonym.synonym_id</td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>synonym_pub_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>tableinfo</h2>

<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>tableinfo_id</td><td>serial</td><td>no</td><td></td><td><li>control.tableinfo_id<li>analysisinput.tableinfo_id<li>processio.inputtable_id<li>projectlink.tableinfo_id<li>magedocumentation.tableinfo_id<li>quantification.resulttable_id</td></tr>
<tr><td>database_id</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>name</td><td>varchar(30)</td><td>no</td><td></td><td></td></tr>
<tr><td>is_updateable</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>modification_date</td><td>date</td><td>no</td><td></td><td></td></tr>
<tr><td>is_view</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>superclass_table_id</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>table_type</td><td>varchar(40)</td><td>no</td><td></td><td></td></tr>
<tr><td>primary_key_column</td><td>varchar(30)</td><td>yes</td><td></td><td></td></tr>
<tr><td>view_on_table_id</td><td>int</td><td>yes</td><td></td><td></td></tr>
<tr><td>is_versioned</td><td>int</td><td>no</td><td></td><td></td></tr>
</table>

<h2>treatment</h2>
ok drop table if exists treatment;
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>treatment_id</td><td>serial</td><td>no</td><td></td><td><li>biomaterialmeasurement.treatment_id</td></tr>
<tr><td>name</td><td>varchar(100)</td><td>yes</td><td></td><td></td></tr>
<tr><td>treatmenttype_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
<tr><td>biomaterial_id</td><td>int</td><td>no</td><td></td><td>biomaterial.biomaterial_id</td></tr>
<tr><td>ordernum</td><td>int</td><td>no</td><td></td><td></td></tr>
<tr><td>protocol_id</td><td>int</td><td>yes</td><td></td><td>protocol.protocol_id</td></tr>
</table>

<h2>wwwuser</h2>
------------------------------ -- f_type -------------------- ------------------------------ ------------------------------ -- fnr_type ------------------ ------------------------------ ------------------------------ -- f_loc --------------------- ------------------------------ ------------------------------ -- fp_key ------------------- ------------------------------  keep track of www users.  this may also be useful  in an audit module at some point (?).
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>email</td><td>varchar(128)</td><td>no</td><td></td><td></td></tr>
<tr><td>username</td><td>varchar(32)</td><td>no</td><td></td><td></td></tr>
<tr><td>wwwuser_id</td><td>serial</td><td>no</td><td></td><td><li>wwwuser_feature.wwwuser_id<li>wwwuser_cvterm.wwwuser_id<li>wwwuser_project.wwwuser_id<li>wwwuser_organism.wwwuser_id<li>wwwuser_genotype.wwwuser_id<li>wwwuser_author.wwwuser_id<li>wwwuser_interaction.wwwuser_id<li>wwwuser_phenotype.wwwuser_id<li>wwwuser_pub.wwwuser_id<li>wwwuser_expression.wwwuser_id<li>wwwuserrelationship.objwwwuser_id<li>wwwuserrelationship.subjwwwuser_id</td></tr>
<tr><td>profile</td><td>text</td><td>yes</td><td></td><td></td></tr>
<tr><td>password</td><td>varchar(32)</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_author</h2>
track wwwuser interest in authors
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>wwwuser_author_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>author_id</td><td>int</td><td>no</td><td></td><td>author.author_id</td></tr>
</table>

<h2>wwwuser_cvterm</h2>
track wwwuser interest in cvterms
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>wwwuser_cvterm_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>cvterm_id</td><td>int</td><td>no</td><td></td><td>cvterm.cvterm_id</td></tr>
</table>

<h2>wwwuser_expression</h2>
track wwwuser interest in expressions
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>expression_id</td><td>int</td><td>no</td><td></td><td>expression.expression_id</td></tr>
<tr><td>wwwuser_expression_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_feature</h2>
track wwwuser interest in features
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>feature_id</td><td>int</td><td>no</td><td></td><td>feature.feature_id</td></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>wwwuser_feature_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_genotype</h2>
track wwwuser interest in genotypes
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>wwwuser_genotype_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>genotype_id</td><td>int</td><td>no</td><td></td><td>genotype.genotype_id</td></tr>
</table>

<h2>wwwuser_interaction</h2>
track wwwuser interest in interactions
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>interaction_id</td><td>int</td><td>no</td><td></td><td>interaction.interaction_id</td></tr>
<tr><td>wwwuser_interaction_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_organism</h2>
track wwwuser interest in organisms
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>organism_id</td><td>int</td><td>no</td><td></td><td>organism.organism_id</td></tr>
<tr><td>wwwuser_organism_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_phenotype</h2>
track wwwuser interest in phenotypes
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>phenotype_id</td><td>int</td><td>no</td><td></td><td>phenotype.phenotype_id</td></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>wwwuser_phenotype_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_project</h2>
link wwwuser accounts to projects
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>project_id</td><td>int</td><td>no</td><td></td><td>project.project_id</td></tr>
<tr><td>wwwuser_project_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuser_pub</h2>
track wwwuser interest in publications
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>pub_id</td><td>int</td><td>no</td><td></td><td>pub.pub_id</td></tr>
<tr><td>wwwuser_pub_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
</table>

<h2>wwwuserrelationship</h2>
track wwwuser interest in other wwwusers
<table border=1><tr><th>Name</th><th>Type</th><th>NULL?</th><th>Comment</th><th>References</th></tr>
<tr><td>wwwuserrelationship_id</td><td>serial</td><td>no</td><td></td><td></td></tr>
<tr><td>world_read</td><td>smallint</td><td>no</td><td></td><td></td></tr>
<tr><td>objwwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
<tr><td>subjwwwuser_id</td><td>int</td><td>no</td><td></td><td>wwwuser.wwwuser_id</td></tr>
</table>

