<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
  <head>
    <title>Examples of Computational Analyses in Chado</title>
  </head>

  <body>
    <h1>Example of Computational Analyses in Chado</h1>

<h2>Background</h2>

    This is meant to illustrate the chado schema.

    <p>

      For a full discussion of the schema, please see the relevant
      directory. For the purposes of illustration, we will assume the
      existence of a DTD or XML-schema derived directly from the
      relational schema. The mapping between these should be fairly
      obvious (hopefully). Note there are a huge number of possible DTDs
      derivable from any relational schema, the implicit DTD presented
      here is not meant to be the canonical one, it's just an expedient
      one for the purposes of illustration.

   <p>
      <a href="../..">Chado companalysis module</a>
   <p>
   <p>
      <a href="../../..">All Chado modules</a>
   <p>

    <p>

      In some cases, joins are flattened - for instance, instead of
      presenting "dbxref"s as elements in their own right, they have
      been combined with the table that has a corresponding foreign key
      index. If it helps, you can imagine the XML as being on top of a
      thin "view layer" on top of the database, in which we are looking
      at the natural join between each table and the dbxref table.

    <p>

      For each state of the database, I'll present the corresponding
      XML, both as standard XML and also as a simple text based tree
      representation. In my opinion, the latter is much better for
      editing and browsing.

    <p>

      I will also present a network-graph image showing the connection
      between the rows in the database at that time state. Every node
      in the graph represents a row. The arcs represent foreign key
      relations. (The graph drawing program renders some of these
      strangely)

    <p>

      <pre>

      Key:

      GREEN ELLIPSE: feature

      CYAN ELLIPSE:  analysisresult

      RED RECTANGLE: analysis

      dotted lines: represent foreign key relationships

      </pre>

    <p>

    <h2>BLAST</h2>

    <h3>Genomic DNA vs Protein</h3>

    This example is a genome 2 protein blastx run, which
    generated two hits against two proteins from swissprot
    for a particular piece of genomic sequence.

    <p>

      You can examine this data as

    <p>
    <table border=1>
        <tr>
          <td>
            <A HREF="blast.itxt">CHADO INDENTED TEXT</A> (with comments)
          </td>
        </tr>
        <tr>
          <td>
            <A HREF="blast.xml">CHADO XML</A> (you may have to view source)
          </td>
        </tr>
    </table>

    <p>
      Or you can visualise it as a PNG:

    <table border=1>
        <tr>
          <td>
            <A HREF="blast.png">PNG</A>
          </td>
        </tr>
    </table>

    <h2>GENSCAN</h2>

    <h3>Gene Prediction</h3>

    This example dataset is a single genscan prediction of two exons
    on a piece of sequence

    <p>

      You can examine this data as

    <p>
    <table border=1>
        <tr>
          <td>
            <A HREF="genscan.itxt">CHADO INDENTED TEXT</A> (with comments)
          </td>
        </tr>
        <tr>
          <td>
            <A HREF="genscan.xml">CHADO XML</A> (you may have to view source)
          </td>
        </tr>
    </table>

    <p>
      Or you can visualise it as a PNG:

    <table border=1>
        <tr>
          <td>
            <A HREF="genscan.png">PNG</A>
          </td>
        </tr>
    </table>

    <h2>CLUSTAL</h2>

    <h3>Multiple Alignment</h3>

    Programs such as clustal produce multiple alignments;
    storing a multiple alignment in chado is similar to
    storing a pairwise alignment; instead of two locations
    per leaf analysisresult, we store N

    <p>

    this example is an alignment of 3 proteins (fly, human and yeast).
    the data is made up. a faked clustal report for this may look like:

    <p>
    <pre>
       +++++     ++++++
    P0 MSGQADIALIGL-AVMG
    P1 MS-QACCCCCGLQAVMC
    P2 MSGQAQQQQQGLQ-VMQ
       +++++     ++++++
    </pre>
    <p>

    there are two conserved blocks between the 3 proteins


    <p>

      You can examine this data as

    <p>
    <table border=1>
        <tr>
          <td>
            <A HREF="clustal.itxt">CHADO INDENTED TEXT</A> (with comments)
          </td>
        </tr>
        <tr>
          <td>
            <A HREF="clustal.xml">CHADO XML</A> (you may have to view source)
          </td>
        </tr>
    </table>

    <p>
      Or you can visualise it as a PNG:

    <table border=1>
        <tr>
          <td>
            <A HREF="clustal.png">PNG</A>
          </td>
        </tr>
    </table>

    <h2>INTERPRO</h2>

    <h3>Prediction domains and families</h3>

    
    <p>

      You can examine this data as

    <p>
    <table border=1>
        <tr>
          <td>
            <A HREF="interpro.itxt">CHADO INDENTED TEXT</A> (with comments)
          </td>
        </tr>
        <tr>
          <td>
            <A HREF="interpro.xml">CHADO XML</A> (you may have to view source)
          </td>
        </tr>
    </table>

    <p>
      Or you can visualise it as a PNG:

    <table border=1>
        <tr>
          <td>
            <A HREF="interpro.png">PNG</A>
          </td>
        </tr>
    </table>

    
    

    <hr>
    <a href="../../../../src/test/perl/perl-docs.html">Chado Tools
      used for producing this document</a>
    <hr>
    <address><a href="mailto:cjm@fruitfly.berkeley.edu">chris mungall</a></address>
<!-- Created: Wed Oct 23 12:38:32 PDT 2002 -->
<!-- hhmts start -->
Last modified: Thu Oct 24 09:14:46 PDT 2002
<!-- hhmts end -->
  </body>
</html>
